### ISSUE TITLE
Update io-write-flow-file.py example

### ISSUE THREAD

**Issue by mhils** [mitmproxy/mitmproxy#6445]
See https://github.com/mitmproxy/mitmproxy/discussions/6444 - the addon should use an option instead of argv.

**Comment by Anu-Ra-g** [comment]
I went through the addon script related to this issue. I wanted to know whether adding an option means using `-w` option or updating the example addon-script to produce the dumpfile **without** the using the `-w` option.

**Comment by Anu-Ra-g** [comment]
```
import random
from typing import BinaryIO

from mitmproxy import http
from mitmproxy import io
from mitmproxy import ctx

from typing import Optional

class Writer:
        
    def load(self, loader):
        loader.add_option(
            name="path",
            typespec=Optional[str],
            default=None,
            help="Specify a path for the mitmproxy dump file"
        )

        if ctx.options.path:
            self.f: BinaryIO = open(ctx.options.path,"wb")
        else:
            self.f: BinaryIO = open("dump.txt","wb")

        self.w = io.FlowWriter(self.f)

    def response(self, flow: http.HTTPFlow) -> None: 
        if random.choice([True, False]):
            self.w.add(flow)

    def done(self):
        self.f.close()

addons = [Writer()]   
```
I looked into the issue. I was able to generate the dump file in the current directory. However I was not able to specify the path in the options, like this 
`mitmproxy -s dumpfile.py --set path=/rough/demo.txt `

**Comment by alpeshjamgade** [comment]
is this already done, if not can i take it up as my first contribution.

**Comment by errorxyz** [comment]
Fixed in #6464
